src/neo-vm/ExecutionEngine.cs

```diff
@@ -42,6 +85,158 @@ public void AddBreakPoint(byte[] script_hash, uint position)
             hashset.Add(position);
         }
 
+        #region Limits
+
+        /// <summary>
+        /// Check if it is possible to overflow the MaxArraySize
+        /// </summary>
+        /// <param name="length">Length</param>
+        /// <returns>Return True if are allowed, otherwise False</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public bool CheckArraySize(int length) => length <= MaxArraySize;
+
+        /// <summary>
+        /// Check if the is possible to overflow the MaxItemSize
+        /// </summary>
+        /// <param name="length">Length</param>
+        /// <returns>Return True if are allowed, otherwise False</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public bool CheckMaxItemSize(int length) => length <= MaxItemSize;
+
+        /// <summary>
+        /// Check if the is possible to overflow the MaxInvocationStack
+        /// </summary>
+        /// <param name="stack">Stack</param>
+        /// <returns>Return True if are allowed, otherwise False</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public bool CheckMaxInvocationStack() => InvocationStack.Count < MaxInvocationStackSize;
+
+        /// <summary>
+        /// Check if the BigInteger is allowed for numeric operations
+        /// </summary>
+        /// <param name="value">Value</param>
+        /// <returns>Return True if are allowed, otherwise False</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public bool CheckBigInteger(BigInteger value) => value.ToByteArray().Length <= MaxSizeForBigInteger;
+
+        /// <summary>
+        /// Check if the BigInteger is allowed for numeric operations
+        /// </summary>
+        /// <param name="byteLength">Value</param>
+        /// <returns>Return True if are allowed, otherwise False</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public bool CheckBigIntegerByteLength(int byteLength) => byteLength <= MaxSizeForBigInteger;
+
+        /// <summary>
+        /// Check if the number is allowed from SHL and SHR
+        /// </summary>
+        /// <param name="shift">Shift</param>
+        /// <returns>Return True if are allowed, otherwise False</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public bool CheckShift(int shift) => shift <= Max_SHL_SHR && shift >= Min_SHL_SHR;
+
+        /// <summary>
+        /// Check if the is possible to overflow the MaxStackSize
+        /// </summary>
+        /// <param name="isStackitemCountStrict">Is stack count strict?</param>
+        /// <param name="count">Stack item count</param>
+        /// <returns></returns>
+        public bool CheckStackSize(bool isStackitemCountStrict, int count = 1)
+        {
+            is_stackitem_count_strict = isStackitemCountStrict;
```

Comment:
I think this line should be:
```suggestion
            is_stackitem_count_strict &= isStackitemCountStrict;
```

Reply:
Here i want to set this var with the same value of `isStackItemCountStrict`